### PR TITLE
use native elasticsearch constructor options

### PR DESCRIPTION
**Pull request by benoitvidis**
# Description

This PR allows to use native Elasticsearch constructor options.
For the time being, the previous way of configuring Elasticsearch is kept in a compatibility layer.

# Related issue

* #772

### CODE REVIEW

File: lib/services/elasticsearch.js [hunk 1]
```diff
@@ -1204,11 +1215,8 @@ function assertNoRouting(esRequest) {
...
  * @throws
  */
-function assertWellFormedRefresh(version, esRequest) {
+function assertWellFormedRefresh(esRequest) {
   if (esRequest.hasOwnProperty('refresh')) {
-    if (compareVersions(version, '5.0') < 0) {
-      throw new BadRequestError(`Refresh parameter is not supported by the version "${version}" of Elasticsearch API`);
-    }
     if (esRequest.refresh !== 'wait_for' && esRequest.refresh !== 'false' && esRequest.refresh !== false) {
       throw new BadRequestError('Refresh parameter only supports the value "wait_for" or false');
```
Comment (kuzzle) [inline lib/services/elasticsearch.js]: MAJOR Merge this if statement with the nested one. [rule](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: lib/services/internalEngine/index.js [hunk 2]
```diff
@@ -79,10 +79,19 @@ class InternalEngine {
...
-        hosts: this.config.hosts || this.config.host + ':' + this.config.port,
-        apiVersion: this.config.apiVersion
-      });
+      let options;
+
+      if (this.config.client) {
+        options = Object.assign({}, this.config.client);
+      }
+      else {
+        options = {
```
Comment (ballinette) [inline lib/services/internalEngine/index.js]: nipticking: add here the same comment as https://github.com/kuzzleio/kuzzle/blob/68a17b30276dd2b1c04f3ae751fe53fcef014a32/lib/services/elasticsearch.js#L1189

Comment (ballinette) [inline lib/services/internalEngine/index.js]: nipticking: add here the same comment as https://github.com/kuzzleio/kuzzle/blob/68a17b30276dd2b1c04f3ae751fe53fcef014a32/lib/services/elasticsearch.js#L1189

File: lib/services/internalEngine/index.js [hunk 3]
```diff
@@ -79,10 +79,21 @@ class InternalEngine {
...
-        apiVersion: this.config.apiVersion
-      });
+      let options;
+
+      if (this.config.client) {
+        options = Object.assign({}, this.config.client);
+      }
+      else {
+        // keep compatibility layer for the time being
+        // @todo: once the install base is clean, remove this
```
Comment (kuzzle) [inline lib/services/internalEngine/index.js]: INFO Complete the task associated to this TODO comment. [rule](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Pull request merged**